spark/src/main/java/org/apache/iceberg/spark/source/IcebergBatchWriter.java

```diff
@@ -269,10 +295,19 @@ public String toString() {
       this.dsSchema = dsSchema;
     }
 
-    @Override
-    public DataWriter<InternalRow> createDataWriter(int partitionId, long taskId, long epochId) {
+    public DataWriter<InternalRow> createWriter(int partitionId, long taskId, long epochId) {
       OutputFileFactory fileFactory = new OutputFileFactory(partitionId, taskId, epochId);
       AppenderFactory<InternalRow> appenderFactory = new SparkAppenderFactory();
+      if (spec.fields().isEmpty()) {
+        return new UnpartitionedWriter(spec, format, appenderFactory, fileFactory, fileIo, targetFileSize);
+      } else {
+        return new PartitionedWriter(spec, format, appenderFactory, fileFactory, fileIo, targetFileSize);
+      }
+    }
+
+    public DataWriter<InternalRow> createWriter(int partitionId, long taskId) {
```

Comment:
Why does this duplicate this method instead of calling `createWriter(partitionId, taskId, 0)`?